components/post_view/combined_system_message/combined_system_message.jsx

```diff
@@ -313,17 +323,23 @@ export default class CombinedSystemMessage extends React.PureComponent {
                     userIds = [this.props.currentUserId];
                 } else {
                     // Not something the current user did or was affected by
-                    continue;
+                    return;
                 }
             }
 
-            content.push(
-                <React.Fragment key={postType + actorId}>
-                    {this.renderFormattedMessage(postType, userIds, actorId)}
-                    <br/>
-                </React.Fragment>
-            );
-        }
+            if (postType === REMOVE_FROM_CHANNEL) {
+                removedUserIds = removedUserIds.concat(userIds);
+            } else if (postType !== REMOVE_FROM_CHANNEL && removedUserIds.length > 0) {
+                content.push(this.renderMessage(postType, removedUserIds, actorId));
+                removedUserIds = [];
+            } else {
+                content.push(this.renderMessage(postType, userIds, actorId));
+            }
+
+            if (removedUserIds.length > 0 && index === messageData.length - 1) {
```

Comment:
This might be a bit clearer if you move this check to be after the loop. You could also switch it back to a for-in loop then since you wouldn't need the index

Reply:
That's true but I ended up placing it here since I need access to `postType` and `actorId` which is convenient to access inside the loop.